api/opentrons/drivers/mag_deck/driver.py

```
@@ -0,0 +1,388 @@
+        '''
+        self.run_flag.wait()
+
+        try:
+            self._send_command(GCODES['PROBE_PLATE'])
+        except (MagDeckError, SerialException, SerialNoResponse) as e:
+            return str(e)
+        self._update_plate_height()
+        return ''
+
+    @property
+    def plate_height(self) -> float:
+        '''
+        Default plate_height for the device is 30;
+        calculated as MAX_TRAVEL_DISTANCE(45mm) - 15mm
+        '''
+        self._update_plate_height()
+        return self._plate_height
+
+    @property
+    def mag_position(self) -> float:
+        '''
+        Default mag_position for the device is 0.0
+        i.e. it boots with the current position as 0.0
+        '''
+        self._update_mag_position()
+        return self._mag_position
+
+    def move(self, distance_mm) -> str:
+        '''
+            Move the magnets along Z axis where the home position is 0.0
+            distance_mm-> a point along Z from 0.0 to plate_height
+        '''
+        self.run_flag.wait()
+
+        try:
+            distance_mm = round(float(distance_mm), GCODE_ROUNDING_PRECISION)
+            self._send_command('{0} Z{1}'.format(GCODES['MOVE'], distance_mm))
+        except (MagDeckError, SerialException, SerialNoResponse) as e:
+            return str(e)
+        self._update_mag_position()
+        return ''
+
+    def get_device_info(self) -> dict:
+        '''
+        Queries Temp-Deck for it's build version, model, and serial number
+
+        returns: dict
+            Where keys are the strings 'version', 'model', and 'serial',
+            and each value is a string identifier
+
+            {
+                'serial': '1aa11bb22',
+                'model': '1aa11bb22',
+                'version': '1aa11bb22'
+            }
+
+        Example input from Temp-Deck's serial response:
+            "serial:aa11bb22 model:aa11bb22 version:aa11bb22"
+        '''
+        if self.simulating:
+            return {
```

This is a mistake I've made in other driver files--we should not return dummy data in simulating mode if possible. We should return empty responses, and then use mocks in test instead of using the dummy values.

---

So, return None?